src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,273 @@
+import { addNewFiatAccount } from 'src/fiatconnect'
+import i18n from 'src/i18n'
+import ForwardChevron from 'src/icons/ForwardChevron'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'FiatDetailsScreen'
+type ScreenProps = StackScreenProps<StackParamList, Screens.FiatDetailsScreen>
+
+type Props = ScreenProps
+
+interface Fields {
+  name: string
+  label: string
+  regex: RegExp
+  placeholderText: string
+  errorMessage: string
+}
+
+// This is a mapping between different fiat account schema to the metadata of the fields that need to be rendered on the bank details screen
+const SCHEMA_TO_FIELD_METADATA_MAP = {
+  AccountNumber: [
+    {
+      name: 'accountName',
+      label: i18n.t('fiatAccountSchema.accountName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
+    },
+    {
+      name: 'institutionName',
+      label: i18n.t('fiatAccountSchema.institutionName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
+    },
+    {
+      name: 'accountNumber',
+      label: i18n.t('fiatAccountSchema.accountNumber.label'),
+      regex: /^[0-9]{10}$/,
+      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
+    },
+  ],
+}
+
+const FiatDetailsScreen = ({ route, navigation }: Props) => {
+  const { t } = useTranslation()
+  const { providerURL, fiatAccountSchema } = route.params
+  const [validInputs, setValidInputs] = useState(false)
+  const [textValue, setTextValue] = useState('')
+  const [errors, setErrors] = useState(new Set<string>())
+  const inputRefs = useRef<string[]>([textValue])
+
+  const dispatch = useDispatch()
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle: i18n.t('fiatDetailsScreen.header'),
+    })
+  }, [navigation])
+
+  const getFieldsBySchema = (fiatAccountSchema: FiatAccountSchema): Fields[] => {
+    return SCHEMA_TO_FIELD_METADATA_MAP[fiatAccountSchema]
+  }
+
+  const formFields = useMemo(() => {
+    const fields = getFieldsBySchema(fiatAccountSchema)
+
+    for (let i = 0; i < fields.length; i++) {
+      inputRefs.current.push('')
+    }
+    return fields
+  }, [fiatAccountSchema])
+
+  const onPressNext = async () => {
+    validateInput()
+
+    if (validInputs) {
+      const body: Record<string, string> = {}
+      for (let i = 0; i < formFields.length; i++) {
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      // TODO: some schema requires additional fields in body in addition
```

Comment:
We'll also need to set the `fiatAccountType` to the [fixed value](https://github.com/fiatconnect/specification/blob/main/fiatconnect-api.md#8321-accountnumber) defined in the spec, depending on the schema being used.